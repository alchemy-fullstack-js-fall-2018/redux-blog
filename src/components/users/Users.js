import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export default class Users extends PureComponent {
  static propTypes = {
    users: PropTypes.array.isRequired,
    fetchUsersPromise: PropTypes.func.isRequired
  };

  fetchUsers = () => {
    this.props.fetchUsersPromise();
  };

  componentDidMount() {
    this.fetchUsers();
  }

  render() {
    const { users } = this.props;
    const usersList = users.map((user, i) => {
      return <li key={i}>{user.name}</li>;
    });

    return (
      <div>
        <ul>
          {users && usersList}
        </ul>
      </div>
    );
  }
}
