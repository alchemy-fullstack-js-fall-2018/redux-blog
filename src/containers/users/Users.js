import { connect } from 'react-redux';
import Users from '../../components/users/Users';
import { fetchUsers } from '../../actions/users';
import { getUsers } from '../../selectors/users';

const mapStateToProps = state => ({
  users: getUsers(state)
});

const mapDispatchToProps = dispatch => ({
  fetchUsers() {
    dispatch(fetchUsers());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Users);
