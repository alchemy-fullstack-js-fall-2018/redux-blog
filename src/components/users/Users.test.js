import Users from './Users';
import { shallow } from 'enzyme';
import React from 'react';
import { fakeUsers } from '../../fixtures/fakeUsers';

describe('Users component', () => {

  it('matches snapshot', () => {
    const users = fakeUsers;
    const fetchUsersPromise = jest.fn();
    const wrapper = shallow(<Users users={users} fetchUsersPromise={fetchUsersPromise} />);
    expect(wrapper).toMatchSnapshot();
  });

});
