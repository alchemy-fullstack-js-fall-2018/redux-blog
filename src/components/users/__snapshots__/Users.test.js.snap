// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component matches snapshot 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      Leanne Graham
    </li>
    <li
      key="1"
    >
      Reanne Graham
    </li>
    <li
      key="2"
    >
      Beanne Graham
    </li>
  </ul>
</div>
`;
